test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala

```diff
@@ -247,202 +280,200 @@ class HelpToSaveConnectorSpec extends ControllerSpecWithGuiceApp with HttpSuppor
+          }
+        }
+      }
+      "the future fails" in {
+        when(GET, storeEmailURL, Map.empty, Map.empty).thenReturn(10, "")
+        val result = connector.storeEmail(email)
+        await(result.value).isLeft shouldBe true
+      }
     }
 
     "getting emails" must {
 
       val email = "email"
 
-      behave like testCommon(
-        mockGet(getEmailURL),
-        () => connector.getEmail(),
-        GetEmailResponse(None),
-        false
-      )
-
       "return a Right if the call comes back with HTTP status 200 and " +
         "valid JSON in the body" in {
-        mockGet(getEmailURL)(Some(HttpResponse(200, Json.toJson(GetEmailResponse(Some(email))), emptyHeaders)))
+        val validResponse = HttpResponse(200, Json.toJson(GetEmailResponse(Some(email))), emptyHeaders)
+        when(GET, getEmailURL, Map.empty, Map.empty).thenReturn(validResponse.status, validResponse.body)
         await(connector.getEmail().value) shouldBe Right(Some(email))
 
-        mockGet(getEmailURL)(Some(HttpResponse(200, Json.toJson(GetEmailResponse(None)), emptyHeaders)))
+        val inValidResponse = HttpResponse(200, Json.toJson(GetEmailResponse(None)), emptyHeaders)
+        when(GET, getEmailURL, Map.empty, Map.empty).thenReturn(inValidResponse.status, inValidResponse.body)
         await(connector.getEmail().value) shouldBe Right(None)
       }
+      "return an error and the call comes back with any other status other than 200" in {
+        Set(500, 303).foreach { status: Int =>
+          whenever(status =!= 200) {
+            val response = HttpResponse(status, "")
+            when(GET, getEmailURL, Map.empty, Map.empty).thenReturn(response.status, response.body)
+            val result = connector.getEmail()
+            await(result.value).isLeft shouldBe true
+          }
+        }
+      }
+      "the future fails" in {
+        when(GET, getEmailURL, Map.empty, Map.empty).thenReturn(0, "")
+        val result = connector.getEmail()
+        await(result.value).isLeft shouldBe true
+      }
+
     }
 
     "getting account-creation-allowed response" must {
-      behave like testCommon(
-        mockGet(accountCreateAllowedURL),
-        () => connector.isAccountCreationAllowed(),
-        Json.toJson(true),
-        true
-      )
 
       "return a Right if the call comes with HTTP 200 and valid response in the body" in {
-        mockGet(accountCreateAllowedURL)(Some(HttpResponse(200, Json.toJson(UserCapResponse()), emptyHeaders)))
-
+        val response = HttpResponse(200, Json.toJson(UserCapResponse()), emptyHeaders)
+        when(GET, accountCreateAllowedURL, Map.empty, Map.empty).thenReturn(response.status, response.body)
         await(connector.isAccountCreationAllowed().value) shouldBe Right(UserCapResponse())
       }
+
+      "return an error and the call comes back with any other status other than 200" in {
+        Set(500, 303).foreach { status: Int =>
+          whenever(status =!= 200) {
```

Comment:
This check is not needed, would always be the case

Reply:
Done